routes/auth.js

```
@@ -0,0 +1,66 @@
+'use strict';
+
+const express = require('express');
+const session = require('express-session');
+const authMiddleware = require('../lib/middleware/authenticate');
+
+const authUrl = process.env.DS_API_OAUTH_URL;
+const webUrl = process.env.WEB_URL || process.env.APP_URL;
+
+module.exports = async () => {
+  const router = express.Router();
+
+  // Wait until we discover OpenID Configuration.
+  const passport = await authMiddleware();
+
+  // Configure sessions & authentication.
+  router.use(
+    session({
+      secret: process.env.APP_SECRET || 'puppet',
```

It might be better to throw if `APP_SECRET` is unset so we don't accidentally use `'puppet'` in prod.